tests/examples/test_examples.py

```diff
@@ -0,0 +1,92 @@
+import glob
+import os
+import subprocess
+import sys
+
+import pytest
+
+NO_EXAMPLES = "no examples found"
+
+
+def can_import_unstructured():
+    try:
+        import unstructured  # noqa: F401
+
+        return True
+    except ImportError:
+        return False
+
+
+get_started_examples = [
+    filename
+    for filename in glob.glob("examples/get_started/**/*.py", recursive=True)
+    # torch-loader will not finish within an hour on Linux runner
+    if "torch" not in filename or os.environ.get("RUNNER_OS") != "Linux"
+]
+
+llm_and_nlp_examples = [
+    filename
+    for filename in glob.glob("examples/llm_and_nlp/**/*.py", recursive=True)
+    # no anthropic token
+    if "claude" not in filename
+    and ("unstructured" not in filename or can_import_unstructured())
+] or [NO_EXAMPLES]
+
+multimodal_examples = [
+    filename
+    for filename in glob.glob("examples/multimodal/**/*.py", recursive=True)
+    # no OpenAI token
+    # and hf download painfully slow
+    if "openai" not in filename and "hf" not in filename
```

Comment:
minor: checks like `"hf" not in filename` a bit concerning (can easily lead to false positives w/o noticing)

optional: I would do a file `.skip.tests`

or we can even put a list of file names right here to be 100% explicit

- first get the list of files
- for each exception `assert filename in the list`
- remove each exception